Cognoma.py

```diff
@@ -0,0 +1,656 @@
+"""
+shared-latent-space/Cognoma.py
+
+This class is an implementation of the abstract class
+DataSetInfoAbstractClass. It contains the specific
+implementations for how to load and visualize the Cognoma
+data set. It saves the visualizations in the /Output/Cognoma
+folder with various parameters of the model in the name
+of the file.
+
+
+Author: Chris Williams
+Date: 6/26/18
+"""
+import os
+import pickle
+import copy
+
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+import pandas as pd
+import plotly as py
+import umap as up
+import plotly.graph_objs as go
+import sklearn as sk
+import scipy
+from scipy import misc
+from sklearn import preprocessing
+from six.moves import cPickle
+
+# Local files
+from DataSetInfoAbstractClass import dataSetInfoAbstract
+from model_objects import model_parameters
+
+
+class dataInfo(dataSetInfoAbstract):
+
+    def __init__(self):
+        """
+        Defines the object's name, file locations, image size and domain names
+
+        Args: None
+
+        Returns: None
+        """
+        self.name = "Cognoma"
+        self.training_file = os.path.join('Data', 'Cognoma_Data', 'Training',
+                                          'Cognoma_Training_Large.pkl')
+        self.rightXDim = 45
+        self.rightYDim = 123
+        self.leftXDim = 100
+        self.leftYDim = 80
+        self.rightDomainName = "Mutation"
+        self.leftDomainName = "Expression"
+
+        # For Cognoma evaluation, which mutation to induce and compare
+        self.mutationID = 485
+
+    def load(self):
+        """
+        Loads the testing and training data from pickle files.
+
+        Args: None
+
+        Returns: (Float array, Float array, Float array, Float array)
+                    The left training data, left testing data,
+                    right training data, right testing data
+        """
+        with open(self.training_file, "rb") as fp:
+            (x_temp, a_temp) = cPickle.load(fp, encoding='bytes')
+
+        # a_temp = preprocessing.scale(a_temp)
+
+        # a_temp = preprocessing.normalize(a_temp, norm='l2')
+
+        x_temp, a_temp = sk.utils.shuffle(x_temp, a_temp)
+
+        length = x_temp.shape[0]
+
+        x_train = x_temp[:int(length * .9)]
+        x_test = x_temp[int(length * .9):]
+
+        a_train = a_temp[:int(length * .9)]
+        a_test = a_temp[int(length * .9):]
+        return (x_train, a_train, x_test, a_test)
+
+    def visualize(self, modelHandle,
```

Comment:
I thought about trying to split it up, but they all use the same data, so you would have to pass a massive amount of arguments around to all of the pieces.

Reply:
You can probably do some amount of breaking up, but I just wasn't sure if it would make it harder to follow because most of it builds off of the previous piece.